composer/trainer/trainer.py

```diff
@@ -628,17 +602,16 @@ def _ddp_reduce_tensor_sum(tensor: Tensor) -> Tensor:
                             self.checkpoint_loader.restore_checkpoint_rng_state(self.state, self.device)
                         continue
 
-                    state.last_batch_size = self._get_batch_size(state.batch)
+                    state.last_batch_size = state.train_data.get_num_samples_in_batch(state.batch)
```

Comment:
as mentioned above, we can remove these from state, and store them in the trainer itself.

Reply:
Same as above.